Add io.github.sigmasd.IRust

sigmaSd opened the pull request:
Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements](https://docs.flathub.org/docs/for-app-authors/requirements) and [App Maintenance](https://docs.flathub.org/docs/for-app-authors/maintanance) pages.
- [x] My pull request follows the instructions at [App Submission](https://docs.flathub.org/docs/for-app-authors/submission).
- [x] I am using only the minimal set of permissions. (If not, please explain each non-standard permission.)
- [x] All assets referenced in the manifest are redistributable by any party. If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. Link:
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. (If not, explain why.)

bbhtt commented:
Please answer the PR template, it is required.

bbhtt commented on io.github.sigmasd.IRust.desktop:
This should be upstream

bbhtt commented on io.github.sigmasd.IRust.metainfo.xml:
This should be upstream

bbhtt commented on io.github.sigmasd.IRust.metainfo.xml:
```diff
@@ -0,0 +1,50 @@
+  <summary>Rust Repl</summary>
+
+  <metadata_license>MIT</metadata_license>
+  <project_license>MIT</project_license>
```
Add developer_name

bbhtt commented on io.github.sigmasd.IRust.svg:
This should be upstream

bbhtt commented on io.github.sigmasd.IRust.yml:
```diff
@@ -0,0 +1,39 @@
+  - org.freedesktop.Sdk.Extension.rust-stable
+
+command: irust
+
```
finish-args is required

hfiguiere replied:
it seems to be terminal only.

hfiguiere replied:
```suggestion
finish-args: []
```

This should be enough to make the linter happy.

CLI application aren't really the ideal use case for flatpak.

sigmaSd replied:
I understand that, my motivation is mostly the exposure the tools get through store, I discovered a lot of cool projects through gnome software.

bbhtt commented on io.github.sigmasd.IRust.metainfo.xml:
```diff
@@ -0,0 +1,50 @@
+  <screenshots>
+    <screenshot type="default">
+      <caption>IRust screenshot</caption>
+      <image>https://github.com/sigmaSd/IRust/blob/13e0f71ec4c3e765ecff3a2e6d1ed900ea0b69f9/irust.png</image>
```
```suggestion
      <image>https://raw.githubusercontent.com/sigmaSd/IRust/13e0f71ec4c3e765ecff3a2e6d1ed900ea0b69f9/irust.png</image>
```

The direct link is needed:

```
appstream-util validate io.github.sigmasd.IRust.metainfo.xml 
io.github.sigmasd.IRust.metainfo.xml: FAILED:
• file-invalid          : <screenshot> failed to load [https://github.com/sigmaSd/IRust/blob/13e0f71ec4c3e765ecff3a2e6d1ed900ea0b69f9/irust.png]
Validation of files failed
```

bbhtt commented on io.github.sigmasd.IRust.metainfo.xml:
```diff
@@ -0,0 +1,50 @@
+    </p>
+  </description>
+
+  <launchable type="desktop-id">IRust.desktop</launchable>
```
```suggestion
  <launchable type="desktop-id">io.github.sigmasd.IRust.desktop</launchable>
```

bbhtt commented:
```
rustfmt is not installed, it's required for beautifying repl code
Do you want IRust to install it? [Y/n]: 
y
rustup is not installed.
rustup is required to install rustfmt
error while installing rustfmt
```

sigmaSd commented:
I'm not sure about the options dependences, I thought I'm going to remove the welcome screen for the flatpak, but maybe it's not that much work to also include the optional dependences

sigmaSd commented:
I addressed all comments, can you please take a second look



> You need to add rustup to the manifest
> 
> ```
> rustfmt is not installed, it's required for beautifying repl code
> Do you want IRust to install it? [Y/n]: 
> y
> rustup is not installed.
> rustup is required to install rustfmt
> error while installing rustfmt
> ```

I disabled the scan for optional dependencies  for now on the flatpak package, I can add those on own my pace later.

hfiguiere commented:
bot, build io.github.sigmasd.IRust

sigmaSd commented on io.github.sigmasd.IRust.yml:
```diff
@@ -0,0 +1,33 @@
+  - name: irust
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 $(which cargo) -t /app/bin/
+      - install -Dm755 $(which rustc) -t /app/bin/
```
I'm copying from the rust stable SDK, my program should continue to work with the recent versions of stable no problem. I see what you mean about controlling the exact version but in practice I can forsee that bundling cargo is going to be a pain , and there is no tangible benefit in my opinion.

hfiguiere commented:
bot, build io.github.sigmasd.IRust

sigmaSd commented:
Should I keep pushing updates (mostly to update the version of irust) or should I wait for this pr to be accepted ?
